src/api-next/api-next.ts

```diff
@@ -1782,4 +1782,98 @@ export class IBApiNext {
         .pipe(map((v: { all: OpenOrder[] }) => v.all))
     );
   }
+
+  /** nextValidId event handler */
+  private readonly onNextValidId = (
+    subscriptions: Map<number, IBApiNextSubscription<number>>,
+    orderId: number
+  ): void => {
+    // this is special to other one-shot callbacks:
+    // we only want to complete one subscription at a time,
+    // to avoid multiple getNextValidOrderId calls to return same value
+    const next = subscriptions.entries().next();
+    if (next && !next.done && next.value[1]) {
+      next.value[1].next({
+        all: orderId,
+      });
+      next.value[1].complete();
+    }
+  };
+
+  /**
+   * Requests the next valid order ID at the current moment.
+   */
+  getNextValidOrderId(): Promise<number> {
+    return lastValueFrom(
+      this.subscriptions
+        .register<number>(
+          () => {
+            this.api.reqIds();
+          },
+          undefined,
+          [[EventName.nextValidId, this.onNextValidId]]
+        )
+        .pipe(map((v: { all: number }) => v.all))
+    );
+  }
+
+  /**
+   * Places or modifies an order.
+   * @param id The order's unique identifier.
+   * Use a sequential id starting with the id received at the nextValidId method.
+   * If a new order is placed with an order ID less than or equal to the order ID of a previous order an error will occur.
+   * @param contract The order's [[Contract]].
+   * @param order The [[Order]] object.
+   */
+  placeOrder(id: number, contract: Contract, order: Order): void {
+    this.api.placeOrder(id, contract, order);
+  }
+
+  /**
+   * Places new order.
+   * Use a sequential id starting with the id received at the nextValidId method.
+   * If a new order is placed with an order ID less than or equal to the order ID of a previous order an error will occur.
+   * @param contract The order's [[Contract]].
+   * @param order The [[Order]] object.
+   *  @see [[getNextValidOrderId]]
+   */
+  placeNewOrder(contract: Contract, order: Order): void {
+    this.getNextValidOrderId().then((orderId: number) => {
+      this.placeOrder(orderId, contract, order);
+    });
+  }
```

Comment:
```suggestion
  async placeNewOrder(contract: Contract, order: Order): Promise<number> {
    const orderId = await this.getNextValidOrderId();
    this.placeOrder(orderId, contract, order);
    return orderId;
  }
```
Let's the return the oder id, otherwise it might be difficult to find it again.